src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -493,22 +493,12 @@ MulticopterPositionControl::limit_altitude(vehicle_local_position_setpoint_s &se
 		return;
 	}
 
-	float altitude_above_home = -(_states.position(2) - _home_pos.z);
+	const float max_local_altitude = _home_pos.z + (-_vehicle_land_detected.alt_max);
```

Comment:
It's still so weird that `land_detected` contains a max altitude :confused:.

Also, this should be `max_z`, right? Because it's negative for up.

Reply:
Yes, it doesn't make any sense to me either having the maximal altitude in the land detector...
You're right altitude would need to be positive. `max_z` is a better name. One moment.

Reply:
Done